tests/test_cache.py

```diff
@@ -415,183 +423,253 @@ def test_method_2(self, dummy_arg):
 
-    Cache entries are considered lingering as long as for_version=True (default).
-    The timeout argument is ignored for lingering entries.
-    Lingering entries are considered stale 60s after their last use.
+    Cache entries will expire 60s after their last access if:
+      - for_version=True (legacy parameter)
+      - refresh_after_access=True
+
+    The timeout argument is ignored for these entries.
     Stale entries are cleaned up on the next call to `clean_stale_entries`.
 
+    Legacy behaviour tests:
     This test checks that this behaviour is consistent for the 4 combinations
-    of (timeout, for_version) via `linger_test_method_1` to `linger_test_method_4`.
+    of (timeout, for_version) via `test_method_1` to `test_method_4`.
+
+    `test_method_5` is used to test that
+        - an entry is properly cleaned up if it is not being used ("initial_read" entry).
+        - the entry's expiry time is properly reset to 60s when it is read.
+
+    New behaviour tests:
+        `test_method_1` and `test_method_3` already test the behaviour for the default
+        value for `refresh_after_access`.
 
-    `linger_test_method_5` is used to test that
-        - a lingering entry is properly cleaned up if it is not being used ("initial_read" entry).
-        - the expiry time of a lingering entry is properly reset to 60s when
-          it is read ("recurring_read" entry).
+        `test_method_22`, `test_method_44` and  `test_method_55` test the new behaviour
+        by explicitly providing the `refresh_after_access` parameter.
 
     """
 
-    class LingeringTest(CacheMissCounter):
+    class RefreshAfterAccessTest(CacheMissCounter):
         @cache
-        def linger_test_method_1(self):
+        def test_method_1(self):
             self.increment_miss_counter()
             return "x1"
 
         @cache(for_version=True)
-        def linger_test_method_2(self):
+        def test_method_2(self):
             self.increment_miss_counter()
             return "x2"
 
         @cache(timeout=1)
-        def linger_test_method_3(self):
+        def test_method_3(self):
             self.increment_miss_counter()
             return "x3"
 
         @cache(timeout=1, for_version=True)
-        def linger_test_method_4(self):
+        def test_method_4(self):
             self.increment_miss_counter()
             return "x4"
 
         @cache
-        def linger_test_method_5(self, dummy_arg: str):
+        def test_method_5(self, dummy_arg: str):
+            self.increment_miss_counter()
+            return dummy_arg
+
+        @cache(refresh_after_access=True)
+        def test_method_22(self):
+            self.increment_miss_counter()
+            return "x2"
+
+        @cache(timeout=1, refresh_after_access=True)
+        def test_method_44(self):
+            self.increment_miss_counter()
+            return "x4"
+
+        @cache(refresh_after_access=True)
+        def test_method_55(self, dummy_arg: str):
```

Comment:
Imo, I think it would be easier to read if the different test methods were named according to what we are trying to test instead of number. However, it could become too long and verbose.